src/js/state/Queries/selectors.ts

```diff
@@ -1,6 +1,37 @@
-import {QueriesState} from "./types"
+import {QueriesState, Query} from "./types"
 import {State} from "../types"
+import TreeModel from "tree-model"
+import {includes} from "lodash"
 
 export default {
-  getRaw: (state: State): QueriesState => state.queries
+  getRaw: (state: State): QueriesState => state.queries,
+  getTags: (state: State): string[] => {
```

Comment:
I'm not sure if you're using this or not in the code above. But if you are, use createSelector from releselect here to derive the state so that the reference of the return value is the same when the args are the same and components don't re-renderer when they don't need to.

Reply:
Same goes for getQueriesByTag below.